hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientReliableTopicProxy.java

```diff
@@ -197,6 +212,114 @@ public LocalTopicStats getLocalTopicStats() {
+                    .map(m -> new ReliableTopicMessage(toData(m), null))
+                    .collect(Collectors.toList());
+            switch (overloadPolicy) {
+                case ERROR:
+                    addAsyncOrFail(payload, returnFuture, messages);
+                    break;
+                case DISCARD_OLDEST:
+                    addAsync(messages, OverflowPolicy.OVERWRITE);
+                    break;
+                case DISCARD_NEWEST:
+                    addAsync(messages, OverflowPolicy.FAIL);
+                    break;
+                case BLOCK:
+                    addAsyncAndBlock(payload, returnFuture, messages, INITIAL_BACKOFF_MS);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unknown overloadPolicy:" + overloadPolicy);
+            }
+        } catch (Exception e) {
+            throw (RuntimeException) peel(e, null,
+                    String.format("Failed to publish messages: %s on topic: %s", payload, getName()));
+        }
+        return returnFuture;
+    }
+
+    private void addAsyncAndBlock(@Nonnull Collection<? extends E> payload, InternalCompletableFuture<Void> returnFuture,
+                                  List<ReliableTopicMessage> messages, long pauseMillis) {
+        ringbuffer.addAllAsync(messages, OverflowPolicy.FAIL).whenCompleteAsync((id, t) -> {
+            if (t != null) {
+                returnFuture.completeExceptionally(t);
+            } else if (id == -1) {
+                getContext().getTaskScheduler().schedule(
+                        () -> addAsyncAndBlock(payload, returnFuture, messages, Math.min(pauseMillis * 2, MAX_BACKOFF)),
+                        pauseMillis, MILLISECONDS);
+            } else {
+                returnFuture.complete(null);
+            }
+        });
+    }
+
+    private void addAsyncOrFail(@Nonnull Collection<? extends E> payload, InternalCompletableFuture<Void> returnFuture,
+                                List<ReliableTopicMessage> messages) {
+        ringbuffer.addAllAsync(messages, OverflowPolicy.FAIL).whenCompleteAsync((id, t) -> {
+            if (t != null) {
+                returnFuture.completeExceptionally(t);
+            } else if (id == -1) {
+                returnFuture.completeExceptionally(new TopicOverloadException(
+                        "Failed to publish messages: " + payload + " on topic:" + getName()));
+            } else {
+                returnFuture.complete(null);
+            }
+        });
+    }
+
+    private InternalCompletableFuture<Void> addAsync(List<ReliableTopicMessage> messages, OverflowPolicy overflowPolicy) {
+        InternalCompletableFuture<Void> returnFuture = new InternalCompletableFuture<>();
+        ringbuffer.addAllAsync(messages, overflowPolicy).whenCompleteAsync((id, t) -> {
+            if (t != null) {
+                returnFuture.completeExceptionally(t);
+            }
+            returnFuture.complete(null);
```

Comment:
Here it's not that critical as on member-side (because of statistics) but move this to an else block.

Reply:
Done